### PR TITLE
[Backport 2.x] [Spotless] Backport of #9 #13 #14 #18 #15 #16 #17spotless PRs 

### PR DESCRIPTION
**Pull request by MitchellGale**
### Description
Backports several spotless PRs to 2.x 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 buildscript {
     ext {
-        opensearch_version = System.getProperty("opensearch.version", "2.10.0-SNAPSHOT")
+        opensearch_version = System.getProperty("opensearch.version", "3.0.0-SNAPSHOT")
```
Comment (MitchellGale) [inline build.gradle:9]:
```suggestion
        opensearch_version = System.getProperty("opensearch.version", "2.10.0-SNAPSHOT")
```

File: build.gradle [hunk 2]
```diff
@@ -120,9 +124,8 @@ allprojects {
         sourceCompatibility = targetCompatibility = "11"
     }
     configurations.all {
-        resolutionStrategy.force "com.squareup.okio:okio:3.5.0"
-        resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib:1.9.0"
-        resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.9.0"
+        resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib:1.6.0"
+        resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib-common:1.6.0"
```
Comment (MitchellGale) [inline build.gradle:128]:
```suggestion
        resolutionStrategy.force "com.squareup.okio:okio:3.5.0"
        resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib:1.9.0"        
        resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.9.0"
```

File: build.gradle [hunk 3]
```diff
@@ -224,5 +228,12 @@ task updateVersion {
         println "Setting version to ${newVersion}."
         // String tokenization to support -SNAPSHOT
         ant.replaceregexp(file:'build.gradle', match: '"opensearch.version", "\\d.*"', replace: '"opensearch.version", "' + newVersion.tokenize('-')[0] + '-SNAPSHOT"', flags:'g', byline:true)
+        ant.replaceregexp(match:'"version": "\\d+.\\d+.\\d+.\\d+', replace:'"version": ' + '"' + newVersion.tokenize('-')[0] + '.0', flags:'g', byline:true) {
+            fileset(dir: projectDir) {
+                include(name: "workbench/package.json")
+                include(name: "workbench/opensearch_dashboards.json")
+            }
+        }
+        ant.replaceregexp(file:'workbench/opensearch_dashboards.json', match:'"opensearchDashboardsVersion": "\\d+.\\d+.\\d+', replace:'"opensearchDashboardsVersion": ' + '"' + newVersion.tokenize('-')[0], flags:'g', byline:true)
```
Comment (MitchellGale) [inline build.gradle:237]:
```suggestion

```

File: common/build.gradle [hunk 4]
```diff
@@ -51,6 +51,9 @@ dependencies {
     testImplementation group: 'com.squareup.okhttp3', name: 'mockwebserver', version: '4.9.3'
 }
 
+checkstyleMain.ignoreFailures = true
+checkstyleTest.ignoreFailures = true
+checkstyleTest.ignoreFailures = true
```
Comment (Yury-Fridlyand) [inline common/build.gradle:56]:
```suggestion
```

File: opensearch/build.gradle [hunk 5]
```diff
@@ -26,6 +26,7 @@ plugins {
     id 'java-library'
     id "io.freefair.lombok"
     id 'jacoco'
+    id 'info.solidsoft.pitest' version '1.9.0'
```
Comment (Yury-Fridlyand) [inline opensearch/build.gradle:29]:
```suggestion
```

**Review by Yury-Fridlyand — COMMENTED** [review]
Execution failed for task ':core:checkstyleTestFixtures'.

File: core/src/main/java/org/opensearch/sql/ast/expression/QualifiedName.java [hunk 6]
```diff
@@ -81,17 +75,12 @@ public Optional<String> first() {
...
-   * For example:
-   * {@code
-   *   QualifiedName name = ...
-   *   Optional<String> first = name.first();
-   *   if (first.isPresent()) {
-   *     name.rest() ...
-   *   }
-   * }
-   * @return  rest part(s)
+   * Get rest parts of the qualified name. Assume that there must be remaining parts so caller is
```
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/ast/expression/QualifiedName.java:78]: :(

File: core/src/test/java/org/opensearch/sql/expression/ReferenceExpressionTest.java [hunk 7]
```diff
@@ -138,58 +137,33 @@ public void array_with_multiple_path_value() {
...
-   *   },
-   *   "address.local": {
-   *     "state": "WA",
-   *   },
-   *   "message": [
-   *     { "info": "message in array" },
-   *     { "info": "Only first index of array used" }
-   *   ]
-   * }
+   * { "name": "bob smith" "project.year": 1990, "project": { "year": 2020 }, "address": { "state":
```
Comment (acarbonetto) [inline core/src/test/java/org/opensearch/sql/expression/ReferenceExpressionTest.java:140]: :(

**Review by acarbonetto — DISMISSED** [review]

File: core/src/test/java/org/opensearch/sql/expression/ReferenceExpressionTest.java (hunk 7 above)
Comment (MitchellGale) [reply core/src/test/java/org/opensearch/sql/expression/ReferenceExpressionTest.java]: Addressed :(, now should be :)

**Pull request closed without merging**